### PR TITLE
Use tasks for downloading and installing packages

### PR DESCRIPTION
**Pull request by poncovka**
Run the `DownloadPackagesTask` task to download the packages.
Run the `InstallPackagesTask` task to install the packages.

### CODE REVIEW

File: pyanaconda/modules/payloads/payload/dnf/dnf_manager.py [hunk 1]
```diff
@@ -286,3 +291,51 @@ def download_packages(self, callback):
...
+        :param display: the DNF progress-reporting object
+        """
+        log.debug("Running the transaction...")
+        exit_reason = None
+
+        try:
+            base.do_transaction(display)
+            exit_reason = "DNF done"
+        except BaseException as e:  # pylint: disable=broad-except
+            log.error('The transaction has ended abruptly: %s', str(e))
```
Comment (jkonecny12) [inline pyanaconda/modules/payloads/payload/dnf/dnf_manager.py:336]: You are mixing `'` and `"`.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me otherwise.

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

File: pyanaconda/modules/payloads/payload/dnf/dnf_manager.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/payloads/payload/dnf/dnf_manager.py]: Fixed.

**Pull request merged**